tools/clang/test/CodeGenSPIRV/spirv.entry-function.unused-param.hlsl

```diff
@@ -0,0 +1,23 @@
+// Run: %dxc -T vs_6_0 -E main
+
+void main (
+// CHECK:  %in_var_A = OpVariable %_ptr_Input_v3float Input
+    in  float3 used_in_var    : A,
+// CHECK:  %in_var_B = OpVariable %_ptr_Input_v3float Input
+    in  float3 unused_in_var  : B,
+// CHECK: %out_var_C = OpVariable %_ptr_Output_v3float Output
+    out float3 used_out_var   : C,
+// CHECK: %out_var_D = OpVariable %_ptr_Output_v3float Output
+    out float3 unused_out_var : D
+) {
+// CHECK:      [[a:%\d+]] = OpLoad %v3float %in_var_A
+// CHECK-NEXT:              OpStore %param_var_used_in_var [[a]]
+// No writing to %param_var_unused_in_var
+// CHECK-NEXT:   {{%\d+}} = OpLoad %v3float %in_var_B
```

Comment:
This load is now redundant?

Reply:
Yep. But it takes some effort to eliminate these unused loads. We right now have `noWriteBack` mechanism in place, but not `noReadingFrom` in `createStageVars` yet. 
https://github.com/Microsoft/DirectXShaderCompiler/blob/92f7997a9db1e302ce405a57c8d199f674cba23b/tools/clang/lib/SPIRV/DeclResultIdMapper.h#L563-L567
It's trivial for spirv-opt to eliminate these unused loads, but not that easy with the stores crossing function boundary. So I didn't do it in this pull request. Maybe a later one if we feel it is necessary.

Reply:
SGTM. Thanks